router/transformer/transformer.go

```diff
@@ -152,17 +152,27 @@ func (trans *handle) Transform(transformType string, transformMessage *types.Tra
 		if err != nil {
 			trans.transformRequestTimerStat.SendTiming(time.Since(s))
 			reqFailed = true
-			trans.logger.Errorf("JS HTTP connection error: URL: %v Error: %+v", url, err)
+			trans.logger.Errorn(
+				"JS HTTP connection error",
+				logger.NewErrorField(err),
+				logger.NewStringField("URL", url),
+			)
 			if retryCount > config.GetInt("Processor.maxRetry", 30) {
 				panic(fmt.Errorf("JS HTTP connection error: URL: %v Error: %+v", url, err))
 			}
 			retryCount++
 			time.Sleep(config.GetDurationVar(100, time.Millisecond, "Processor.retrySleep", "Processor.retrySleepInMS"))
 			// Refresh the connection
+			httputil.CloseResponse(resp)
```

Comment:
possible to move this logic out of for loop and create a function, we can then add a defer statement to close response just after opening reader for the response? explicitly closing response might result in miss in one place or other.

Reply:
Yeah I agree.
In this case I went against that thought though so this would be easy(and quick) to review.
Can take up the suggested refactor at a later point.